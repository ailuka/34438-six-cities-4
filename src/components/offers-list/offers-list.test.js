import React from "react";
import renderer from "react-test-renderer";
import OffersList from "./offers-list.jsx";

const offers = [
  {
    id: `offer-1`,
    image: `http://lorempixel.com/400/200/`,
    price: 120,
    rating: 80,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    isBookmarked: false,
    isPremium: true,
  },
  {
    id: `offer-2`,
    image: `http://lorempixel.com/400/200/`,
    price: 80,
    rating: 80,
    name: `Wood and stone place`,
    type: `Private room`,
    isBookmarked: true,
    isPremium: false,
  },
  {
    id: `offer-3`,
    image: `http://lorempixel.com/400/200/`,
    price: 132,
    rating: 80,
    name: `Canal view Prinsengracht`,
    type: `Apartment`,
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: `offer-4`,
    image: `http://lorempixel.com/400/200/`,
    price: 180,
    rating: 100,
    name: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    isBookmarked: false,
    isPremium: true,
  }
];

describe(`src/offers-list.jsx`, () => {
  describe(`when the offers is a non-empty array`, () => {
    it(`should render with data`, () => {
      const tree = renderer.create(
          <OffersList
            offers={offers}
            onOfferDetailsOpen={() => null}
          />
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe(`when the offers is an empty array`, () => {
    it(`should render with no data`, () => {
      const tree = renderer.create(
          <OffersList
            offers={[]}
            onOfferDetailsOpen={() => null}
          />
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
