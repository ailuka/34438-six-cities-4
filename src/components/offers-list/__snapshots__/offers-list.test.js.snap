// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffersList snapshot when the offers list is on details page should render with the correct class 1`] = `
<div
  className="near-places__list places__list"
>
  <article
    className="near-places__card place-card"
    id="offer-1"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="img/room.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className=" place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            120
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          Beautiful & luxurious apartment at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="near-places__card place-card"
    id="offer-2"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/2"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="img/apartment-02.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className=" place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            80
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/2"
          onClick={[Function]}
        >
          Wood and stone place
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Room
      </p>
    </div>
  </article>
  <article
    className="near-places__card place-card"
    id="offer-3"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/3"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="img/apartment-03.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className=" place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            132
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/3"
          onClick={[Function]}
        >
          Canal view Prinsengracht
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="near-places__card place-card"
    id="offer-4"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/4"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="img/apartment-01.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className=" place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            180
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/4"
          onClick={[Function]}
        >
          Nice, cozy, warm big bed apartment
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
</div>
`;

exports[`OffersList snapshot when the offers list is on main page should render with the correct class 1`] = `
<div
  className="cities__places-list tabs__content places__list"
>
  <article
    className="cities__place-card place-card"
    id="offer-1"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="img/room.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className=" place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            120
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          Beautiful & luxurious apartment at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    id="offer-2"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/2"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="img/apartment-02.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className=" place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            80
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/2"
          onClick={[Function]}
        >
          Wood and stone place
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Room
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    id="offer-3"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/3"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="img/apartment-03.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className=" place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            132
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/3"
          onClick={[Function]}
        >
          Canal view Prinsengracht
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    id="offer-4"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/4"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="img/apartment-01.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className=" place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            180
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/4"
          onClick={[Function]}
        >
          Nice, cozy, warm big bed apartment
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
</div>
`;
